spot_wrapper/spot_dance.py

```diff
@@ -0,0 +1,83 @@
+import time
+
+from bosdyn.choreography.client.choreography import (
+    load_choreography_sequence_from_txt_file,
+)
+from bosdyn.client import ResponseError
+from bosdyn.client.exceptions import UnauthenticatedError
+from bosdyn.client.lease import LeaseKeepAlive
+from bosdyn.client.robot import Robot
+import typing
+
+
+class SpotDance:
+    def __init__(
+        self,
+        robot: Robot,
+        choreography_client: typing.Any,
+        lease_client: typing.Any,
+        is_licensed_for_choreography: typing.Any,
+    ):
+        self._robot = robot
+        self._choreography_client = choreography_client
+        self._lease_client = lease_client
+        self._is_licensed_for_choreography = is_licensed_for_choreography
+
+    def execute_dance(self, filepath: str) -> tuple[bool, list]:
+        if not self._is_licensed_for_choreography:
+            return False, "Robot is not licensed for choreography."
+        lease = self._lease_client.acquire()
+        lk = LeaseKeepAlive(self._lease_client)
```

Comment:
Is there a reason you need the lease_client or to create a LeaseKeepAlive? This seems like it would conflict with SpotWrapper getting a lease

Reply:
I would recommend having a dance function in SpotWrapper, put the decorator on that, and internal to that running your function from spot_dance. This way we avoid the circular import and also avoid trying to get a lease in multiple places.